### PR TITLE
Fix bug when sanitizing empty body

### LINKED ISSUE #13510: [BUG] Recording framework isn't correctly recording Content-Length of Head response
In [this test](https://github.com/Azure/azure-sdk-for-net/pull/12400/files?file-filters%5B%5D=.cs#diff-7f7fefef0eb98fcf1b3e66bc27fd93faR1075), the 2nd to last call is GetProperties().  Its response is supposed to contain the header "Content-Length=1024" (this is what happens in live mode).  Unfortunately, it is being [recorded as 0](https://github.com/Azure/azure-sdk-for-net/pull/12400/files?file-filters%5B%5D=.json#diff-1c90591f6e2c53c82c658a767548af18).  (Line 2292, GitHub won't load the link I tried to create to that specific line.)

I believe this is because the response doesn't have a body, and the recording framework is setting Content-Length = 0 base on the body length, ignoring the actual header.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Sanitizing response from a HEAD request should not result in resetting content-length to 0.

Fixes https://github.com/Azure/azure-sdk-for-net/issues/13510

### CODE REVIEW

**Comment by pakrym** [comment]
Needs test and I don't understand what the issue was? Are we sending an empty body with non-zero CL?

**Comment by pakrym** [comment]
Ah, it was a HEAD request. We should that for filtering instead.

**Comment by JoshLove-msft** [comment]
> Ah, it was a HEAD request. We should that for filtering instead.

What if a head request somehow returns a body and we therefore skip sanitizing it? This seems safer.

**Comment by pakrym** [comment]
>	What if a head request somehow returns a body and we therefore skip sanitizing it? Or what if a get response has a non-zero CL and no body?

That would be HTTP spec violation

**Comment by JoshLove-msft** [comment]
> > What if a head request somehow returns a body and we therefore skip sanitizing it? Or what if a get response has a non-zero CL and no body?
> 
> That would be HTTP spec violation

Yeah..

File: sdk/core/Azure.Core/tests/RecordSessionTests.cs [hunk 1]
```diff
@@ -409,6 +409,39 @@ public void RecordSessionLookupSkipsRequestBodyWhenFilterIsOn()
             Assert.Throws<TestRecordingMismatchException>(() => playbackTransport.Process(message));
         }
 
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public void ContentLengthUpdatedCorrectlyOnEmptyBody(bool isHeadRequest)
```
Comment (pakrym) [inline sdk/core/Azure.Core/tests/RecordSessionTests.cs:415]: extra super nit: I would not make this test parameterized and instead had two separate ones.

**Review by pakrym — APPROVED** [review]

File: sdk/core/Azure.Core/tests/RecordSessionTests.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/core/Azure.Core/tests/RecordSessionTests.cs:415]: I can make two separate test wrappers and leave this as an internal helper.

**Pull request merged**